src/utils/postbox/postboxWrapperV2.ts

```diff
@@ -0,0 +1,1097 @@
+    postRestriction?: PostRestriction,
+  ) => Promise<Result<string>>;
+
+  // For a given topic ID
+  getTopicData?: (
+    topicId: number,
+    collectionId: PublicKey,
+  ) => Promise<Result<ForumPost>>;
+
+  // Create a post
+  createForumPost: (
+    post: {
+      subj?: string;
+      body: string;
+      meta?: any;
+    },
+    topicId: number,
+    collectionId: PublicKey,
+  ) => Promise<Result<string>>;
+
+  editForumPost: (
+    collectionId: PublicKey,
+    post: ForumPost,
+    newPostData: {
+      subj?: string;
+      body: string;
+      meta?: any;
+    },
+  ) => Promise<Result<string>>;
+
+  // For a given topic, the messages
+  getTopicMessages: (
+    topicId: number,
+    collectionId: PublicKey,
+  ) => Promise<Result<ForumPost[]>>;
+
+  deleteForumPost: (
+    forumPost: ForumPost,
+    collectionId: PublicKey,
+    asMod?: boolean,
+  ) => Promise<Result<string>>;
+
+  // Vote a post up
+  voteUpForumPost: (
+    post: ForumPost,
+    collectionId: PublicKey,
+  ) => Promise<Result<string>>;
+
+  // Vote a post down
+  voteDownForumPost: (
+    post: ForumPost,
+    collectionId: PublicKey,
+  ) => Promise<Result<string>>;
+
+  // This is the same as createPost, but additionally,
+  // post.parent = postId
+  replyToForumPost: (
+    replyToPost: ForumPost,
+    collectionId: PublicKey,
+    post: {
+      subj?: string;
+      body: string;
+      meta?: any;
+    },
+  ) => Promise<Result<string>>;
+
+  // For a given topic, the messages
+  getReplies: (
+    topic: ForumPost,
+    collectionId: PublicKey,
+  ) => Promise<Result<ForumPost[]>>;
+
+  getForumPostRestriction: (
+    collectionId: PublicKey,
+  ) => Promise<Result<PostRestriction | null>>;
+
+  setForumPostRestriction: (
+    collectionId: PublicKey,
+    restriction: PostRestriction,
+  ) => Promise<Result<string>>;
+
+  deleteForumPostRestriction: (
+    collectionId: PublicKey,
+  ) => Promise<Result<string>>;
+
+  canCreateTopic: (collectionId: PublicKey) => Promise<boolean>;
+
+  canPost: (collectionId: PublicKey, topic: ForumPost) => Promise<boolean>;
+
+  canVote: (
+    collectionId: PublicKey,
+    post: ForumPost,
+  ) => Promise<Result<boolean>>;
+
+  getVote: (
+    collectionId: PublicKey,
+    post: ForumPost,
+  ) => Promise<Result<boolean>>;
+
+  getVotes: (collectionId: PublicKey) => Promise<Result<ChainVoteEntry[]>>;
+
+  getNFTsForCurrentUser: () => Promise<Result<PublicKey[]>>;
+
+  getNFTMetadataForCurrentUser: () => Promise<
+  Promise<Result<DisplayableToken>>[] | DispatchError
+  >;
+
+  transferNFTs: (
+    receiverId: PublicKey,
+    mint: PublicKey,
+    sendTransaction: (
+      transaction: Transaction,
+      connection: Connection,
+    ) => Promise<string>,
+  ) => Promise<Result<string>>;
+}
+
+export class DispatchForumV2 implements IForum {
+  public user: User;
+
+  constructor(wallet: WalletInterface, conn: Connection, cluster: Cluster) {
+    this.user = new User(wallet, conn, cluster);
```

Comment:
main place to look at here, all initialization happens within the `User` object, so the wrapper can remain a client wrapper.

Reply:
do we need to give the cluster here rather than wrapping it within conn?

Reply:
see above comment about passing in as option in DispatchConnection